src/callbacks_step/ape_euler_coupling.jl

```diff
@@ -0,0 +1,111 @@
+# TODO: Allow kwargs that are later used when constructing `integrator_euler`?
+mutable struct ApeEulerCouplingCallback{RealT<:Real}
+  stepsize_callback_ape::StepsizeCallback{RealT}
+  stepsize_callback_euler::StepsizeCallback{RealT}
+  averaging_callback::DiscreteCallback{<:Any, <:AveragingCallback}
+  integrator_euler::Union{Nothing, AbstractODEIntegrator}
+end
```

Comment:
Sounds reasonable to me then. Unless @ranocha thinks it's a bad idea, leave it be as it is for now.

Reply:
1. Make it work
2. Make it fast

I don't know in which state you are right now, but this approach seems to be okay for stage 1. We should add some timings for this stuff to decide whether we need to change something in stage 2.

Reply:
I restructured the coupling callback in 295c1fe, so that all information to construct the Euler integrator is available during the constructor call now and these `Union` shenanigans aren't necessary anymore.